codalab_service.py

```diff
@@ -579,21 +616,31 @@ def _run_compose_cmd(self, cmd):
         )
         compose_env_string = ' '.join('{}={}'.format(k, v) for k, v in self.compose_env.items())
         print('(cd {}; {} {})'.format(self.compose_cwd, compose_env_string, command_string))
-        if not self.args.dry_run:
+        if self.args.dry_run:
+            success = True
+        else:
             try:
                 popen = subprocess.Popen(
                     command_string,
                     cwd=self.compose_cwd,
                     env=self.compose_env,
                     shell=True,
                     stdout=subprocess.PIPE,
-                    stderr=subprocess.PIPE,
+                    stderr=subprocess.STDOUT,
                 )
-                for stdout_line in popen.stdout:
+                # Note: don't do `for stdout_line in popen.stdout` because that buffers.
```

Comment:
Looks good. What exactly was the issue with using `stderr=subprocess.STDOUT` and `for stdout_line in popen.stdout`?

Reply:
It would just buffer the output, so that the output wouldn't show up line by line.